crates/chainspec/src/api.rs

```diff
@@ -38,7 +39,7 @@ pub trait EthChainSpec: Send + Sync + Unpin + Debug {
     fn prune_delete_limit(&self) -> usize;
 
     /// Returns a string representation of the hardforks.
-    fn display_hardforks(&self) -> impl Display;
+    fn display_hardforks(&self) -> DisplayHardforks;
```

Comment:
the `DisplayHardforks` type is restricted to forkcondition and the `Hardforks` trait, which I'd like not to integrate in this chainspec trait.

this call is only ever called once on launch and I'd make this a `Box<dyn Display>` instead

Reply:
this change can also be submitted individually

Reply:
let's do issue also, to not forget that